src/python/controller/extract/extract.py

```
@@ -46,40 +49,27 @@ def verify_archive(archive_path: str):
-                # Check magic bytes
-                header = f.read(8)
-                recognized = False
-                for signature, name in _ARCHIVE_SIGNATURES:
-                    if header[:len(signature)] == signature:
-                        recognized = True
-                        break
-
-                if not recognized:
-                    # Unrecognized format — skip verification
-                    return
+            result = subprocess.run(
```

Ruff (0.15.4)

[error] `subprocess` call: check for execution of untrusted input

(S603)